components/strava/strava.app.js

```diff
@@ -31,5 +46,56 @@ module.exports = {
         })
       ).data;
     },
+    async createNewActivity($, data) {
+      return (
+        await this._makeAPIRequestWithPlatform($, {
+          method: "POST",
+          path: "/activities",
+          data,
+        })
+      );
+    },
+    async getActivityById($, data) {
+      const {
+        id,
+        ...dataRest
+      } = data;
+      return (
+        await this._makeAPIRequestWithPlatform($, {
+          method: "GET",
+          path: `/activities/${id}${this.makeQueryString(dataRest)}`,
+        })
+      );
+    },
+    async listActivities($, data) {
+      return (
+        await this._makeAPIRequestWithPlatform($, {
+          method: "GET",
+          path: `/activities${this.makeQueryString(data)}`,
+        })
+      );
+    },
+    async updateActivity($, data) {
+      const {
+        id,
+        ...dataRest
```

Comment:
Can the user pass an `id` property in the update request? i.e. could that clobber the `id` property you pass here, that you use as the activity ID?

Reply:
updated to activityId